src/shared/components/FormComponents/EditorController/EditorController.tsx

```diff
@@ -46,57 +33,24 @@ export const EditorController = <T extends FieldValues>({
       <Controller
         name={name}
         control={control}
-        render={({ field: { onChange, value }, fieldState: { error } }) => {
-          const handleDebouncedChange = debounce(onChange, CHANGE_DEBOUNCE_VALUE);
-          const handleBlur = () => {
-            if (withDebounce) {
-              handleDebouncedChange.flush();
-            }
-          };
-          const handleChange = withDebounce ? handleDebouncedChange : onChange;
-
-          return (
-            <StyledFlexColumn sx={{ position: 'relative' }} data-testid={dataTestid}>
-              <StyledMdEditor
-                sanitize={(content: string) => getSanitizedContent(content)}
-                editorId={editorId}
-                className={`${uiType} ${disabled ? 'disabled' : ''} ${error ? 'has-error' : ''}`}
-                ref={editorRef}
-                modelValue={value ?? ''}
-                onChange={handleChange}
-                onBlur={handleBlur}
-                language={LANGUAGE_BY_DEFAULT}
-                disabled={disabled}
-                placeholder={t('textPlaceholder')}
-                defToolbars={getDefToolbars({
-                  onInsert,
-                  onChange: handleChange,
-                  setFileSizeExceeded,
-                  setIncorrectFormat: setIncorrectFileFormat,
-                  setIsLoading,
-                })}
-                customIcon={getCustomIcons()}
-                toolbars={getToolbars()}
-                footers={[]}
-              />
-              <StyledFlexSpaceBetween sx={{ m: theme.spacing(0.4, 0, 2) }}>
-                <FooterMessage
-                  inputSize={(value ?? '').length}
-                  key="footer-message"
-                  error={error}
-                />
-                {!error?.message && (
-                  <CharacterCounter
-                    inputSize={(value ?? '').length}
-                    disabled={disabled}
-                    key="character-counter"
-                  />
-                )}
-              </StyledFlexSpaceBetween>
-              {isLoading && <Spinner uiType={SpinnerUiType.Secondary} />}
-            </StyledFlexColumn>
-          );
-        }}
+        render={({ field: { onChange, value }, fieldState: { error } }) => (
+          <Editor
+            editorId={editorId}
+            editorRef={editorRef}
+            value={value}
+            onChange={onChange}
+            onInsert={onInsert}
+            onFileExceeded={setFileSizeExceeded}
+            onIncorrectFileFormat={setIncorrectFileFormat}
+            onLoading={setIsLoading}
```

Comment:
maybe move isLoading and setIsLoading to the Editor component?

Reply:
done